code/modules/modular_computers/file_system/programs/engineering/power_monitor.dm

```diff
@@ -84,32 +84,94 @@
 		ui.set_auto_update(1)
 
 // Refreshes list of active sensors kept on this computer.
-/datum/nano_module/power_monitor/proc/refresh_sensors()
+/datum/nano_module/program/power_monitor/proc/refresh_sensors()
 	grid_sensors = list()
 	var/connected_z_levels = GetConnectedZlevels(get_host_z())
 	for(var/obj/machinery/power/sensor/S in SSmachines.machinery)
 		if(get_z(S) in connected_z_levels) // Consoles have range on their Z-Level. Sensors with long_range var will work between Z levels.
 			grid_sensors += S
-			GLOB.destroyed_event.register(S, src, /datum/nano_module/power_monitor/proc/remove_sensor)
+			GLOB.destroyed_event.register(S, src, /datum/nano_module/program/power_monitor/proc/remove_sensor)
 
-/datum/nano_module/power_monitor/proc/remove_sensor(var/removed_sensor, var/update_ui = TRUE)
+/datum/nano_module/program/power_monitor/proc/remove_sensor(var/removed_sensor, var/update_ui = TRUE)
 	if(active_sensor == removed_sensor)
 		active_sensor = null
 		if(update_ui)
 			SSnano.update_uis(src)
 	grid_sensors -= removed_sensor
-	GLOB.destroyed_event.unregister(removed_sensor, src, /datum/nano_module/power_monitor/proc/remove_sensor)
+	GLOB.destroyed_event.unregister(removed_sensor, src, /datum/nano_module/program/power_monitor/proc/remove_sensor)
+
+/datum/nano_module/program/power_monitor/proc/is_sysadmin(var/mob/user)
+	if(program)
+		return program.can_run(user, program.computer.get_network())
+	return FALSE
 
 // Allows us to process UI clicks, which are relayed in form of hrefs.
-/datum/nano_module/power_monitor/Topic(href, href_list)
+/datum/nano_module/program/power_monitor/Topic(href, href_list, state)
 	if(..())
 		return 1
+
+	var/mob/user = usr
+
 	if( href_list["clear"] )
 		active_sensor = null
 		. = 1
 	if( href_list["refresh"] )
 		refresh_sensors()
 		. = 1
+
+	if(href_list["toggle_breaker"])
+		var/obj/machinery/power/apc/A = locate(href_list["toggle_breaker"])
+
+		if(!CanInteract(user, state) || QDELETED(A))
+			return 0
+
+		A.operating = !A.operating
```

Comment:
There's a proc for this. It's pretty bad just setting vars on APCs, easy to break stuff.

Reply:
Power channels are promptly cleaned up by the update proc, shouldn't be breaking anything.

Reply:
The update proc need not be called every tick. There's a proc for this; use it.